api/v1beta1/cloudstackcluster_types.go

```diff
@@ -101,7 +101,7 @@ type CloudStackClusterSpec struct {
 
 	// CloudStack domain.
 	// +optional
-	Domain string `json:"domain,omitempty"`
+	Domain CloudStackResourceIdentifier `json:"domain,omitempty"`
```

Comment:
this will be a backwards incompatible change, right? How can we start making these schema changes backwards compatible to decouple ourselves from downstream consumers?